checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakTransfer.java

```diff
@@ -0,0 +1,139 @@
+      TernaryExpressionNode node, TransferInput<CFValue, CFStore> input) {
+    TransferResult<CFValue, CFStore> result = super.visitTernaryExpression(node, input);
+    updateStoreWithTempVar(result, node);
+    return result;
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitMethodInvocation(
+      final MethodInvocationNode node, final TransferInput<CFValue, CFStore> input) {
+
+    TransferResult<CFValue, CFStore> result = super.visitMethodInvocation(node, input);
+
+    handleCreatesMustCallFor(node, result);
+    updateStoreWithTempVar(result, node);
+
+    Node receiver = node.getTarget().getReceiver();
+    // If there is a temporary variable for the receiver, update its type.
+    MustCallAnnotatedTypeFactory mcAtf =
+        rlTypeFactory.getTypeFactoryOfSubchecker(MustCallChecker.class);
+    Node accumulationTarget = mcAtf.getTempVar(receiver);
+
+    if (accumulationTarget != null) {
+      String methodName = node.getTarget().getMethod().getSimpleName().toString();
+      methodName = rlTypeFactory.adjustMethodNameUsingValueChecker(methodName, node.getTree());
+      accumulate(accumulationTarget, result, methodName);
+    }
+
+    return result;
+  }
+
+  /**
+   * Clears the called-methods store of all information about the target if an @CreatesMustCallFor
+   * method is invoked and the type factory can create obligations. Othewise, does nothing.
+   *
+   * @param n a method invocation
+   * @param result the transfer result whose stores should be cleared of information
+   */
+  private void handleCreatesMustCallFor(
+      MethodInvocationNode n, TransferResult<CFValue, CFStore> result) {
+    if (!rlTypeFactory.canCreateObligations()) {
+      return;
+    }
+
+    List<JavaExpression> targetExprs =
+        CreatesMustCallForElementSupplier.getCreatesMustCallForExpressions(
+            n, rlTypeFactory, rlTypeFactory);
+    for (JavaExpression targetExpr : targetExprs) {
+      AnnotationMirror defaultType = rlTypeFactory.top;
+      if (result.containsTwoStores()) {
+        CFStore thenStore = result.getThenStore();
+        thenStore.clearValue(targetExpr);
+        thenStore.insertValue(targetExpr, defaultType);
+        CFStore elseStore = result.getElseStore();
+        elseStore.clearValue(targetExpr);
+        elseStore.insertValue(targetExpr, defaultType);
+      } else {
+        CFStore store = result.getRegularStore();
+        store.clearValue(targetExpr);
+        store.insertValue(targetExpr, defaultType);
+      }
+    }
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitObjectCreation(
+      ObjectCreationNode node, TransferInput<CFValue, CFStore> input) {
+    TransferResult<CFValue, CFStore> result = super.visitObjectCreation(node, input);
+    updateStoreWithTempVar(result, node);
+    return result;
+  }
+
+  /**
+   * This method either creates or looks up the temp var t for node, and then updates the store to
```

Comment:
What is "the temp var"?  This needs either an explanation of the implementation approach, or a pointer to that explanation.